### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by wintan1418**
The App component is a  stateful parent component.
 I have made a class component. Initialized its state in the constructor with three properties: total, next and operation.
just like in the calculate.js module. By default, they should all be set to null.
Implemented a render function that should display Display and ButtonPanel.
Implemented the handleClick function which takes the button name as an argument and changes the state based on the result of the calculate function.
Passed the total or next value to Display the current result.
Passed handleClick as a click handler property to ButtonPanel.

Implemented the handleClick function which takes button name as an argument and returns clickHandler from props.
Passed the clickHandler property to each button in the panel.

Implemented the handleClick function which takes button name as an argument and returns clickHandler from props.
Used the onClick event to attach the handleClick event to the button.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      result = big1.times(big2);
+      break;
+    case '÷':
+      if (big2.toNumber() === 0) {
+        return 'Error alert!: unreasonable division by zero.';
+      }
+      result = big1.div(big2);
+      break;
+    default:
+      throw new Error(`Wrong operator ${op}`);
```
Comment (george-shammar) [inline src/logic/operate.js:25]: Nice effort so far. :+1: You've put in a lot work into this.

- [ ] The `division`, `%` operators generate the error below. 
error

- [ ]  The `X` operator does not function at all.

File: src/component/App.js [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  get result() {
+    const { next, total, operation } = this.state;
+
+    return `${next || ''}${total || ''}${operation || ''}`;
+  }
+
```
Comment (george-shammar) [inline src/component/App.js:24]: - [ ] The way this prop is coded and passed to the Display component is producing an awkward output. Please see below. What I tried to do was add two numbers together.
Screenshot 2021-08-02 at 12-37-05 React App
You have something like the image above for all your calculator operations. Buttons are concatenated with the operations.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @wintan1418 , 

Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there! :+1: 


Your calculator is not working as it should. This is the last milestone but unfortunately, most of the required changes are carry overs from the previous milestones. Perhaps you broke your code and created new bugs in the process of implementing this milestone. Please see below:


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @wintan1418**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**